src/documentation/search/SearchResultList.tsx

```diff
@@ -38,37 +48,51 @@ const SearchResultList = ({
 interface SearchResultItemProps {
   value: Result;
   onClose: () => void;
+  viewedResults: string[];
+  onViewResult: (id: string) => void;
 }
 
 const SearchResultItem = ({
-  value: { extract, navigation, containerTitle, title },
+  value: { extract, navigation, containerTitle, title, id },
   onClose,
+  viewedResults,
+  onViewResult,
 }: SearchResultItemProps) => {
   const [, setState] = useRouterState();
+  const query = getQueryString(navigation);
+  const url =
+    window.location.toString().split("?")[0] + (query ? "?" + query : "");
+
   return (
-    <Stack>
-      <Stack px={8} py={2} spacing={0}>
-        {title !== containerTitle && (
-          <Text fontSize="sm" color="gray.600" fontWeight="bold">
-            {containerTitle}
-          </Text>
-        )}
-        <Link
-          onClick={(e) => {
-            e.preventDefault();
-            onClose();
-            setState(navigation);
-          }}
-        >
+    <Stack pl="3px" pr="3px">
+      <Link
+        bgColor={viewedResults.includes(id) ? "#efedf5" : "unset"}
+        borderRadius="md"
+        href={url}
+        onClick={(e) => {
+          e.preventDefault();
+          onClose();
+          setState(navigation);
+          onViewResult(id);
```

Comment:
Maybe the setState call should move up to inside the handler passed for onViewResult?

Reply:
[`f2e178f`](https://github.com/microbit-foundation/python-editor-next/pull/477/commits/f2e178f02edf9d9bf73e5c8a243016d57fb74dba). We now handle closing the on in handleViewResult as well, so this no longer has to be passed down.